### DISCUSSION TITLE
DEPRECATION: trivy client

### DISCUSSION THREAD

**Discussion by knqyf263** [aquasecurity/trivy#2119] [Announcements]
## Summary
Please migrate `trivy client` to `trivy image --server`.

### Before
```
$ trivy server
2022-05-15T14:25:58.414+0300    INFO    Listening localhost:4954...
```

```
$ trivy client [YOUR_IMAGE]
```

### After
```
$ trivy server
2022-05-15T14:25:58.414+0300    INFO    Listening localhost:4954...
```

```
$ trivy image --server http://localhost:4954 [YOUR_IMAGE]
```

## Due date
December 31, 2022

## Description
Trivy used to support container images only, but it now supports filesystem as well. Then, filesystem scanning also added support for client/server mode like `trivy fs --server [YOUR_TRIVY_SERVER] /path/to/your_project`. Accordingly, image scanning also migrated from `trivy client` to `trivy image --server`.

`trivy client` will also be available through 2022, after which it will be removed. Please migrate to `trivy image --server` as soon as possible.

**Comment by cm3brian** [comment]
Is there any chance that the env var `TRIVY_REMOTE` could still be used and if set to just automatically set `--server [YOUR_TRIVY_SERVER]` globally? seems a little weird to remove that ability as "forward progress"? environment vars in pipelines with docker containers is very common and this kind of feels a little backwards that I now need to hard code each call with this parameter?

**Reply by knqyf263** [reply]
You can use `TRIVY_SERVER` under each subcommand.

**Reply by cm3brian** [reply]
This works, thank you. Hopefully the env vars can be documented somewhere for these, very helpful to know.

**Reply by knqyf263** [reply]
https://aquasecurity.github.io/trivy/v0.34/docs/references/customization/envs/
